src/imgix-core-js.js

```diff
@@ -124,6 +124,74 @@
       }
     };
 
+    ImgixClient.prototype.buildSrcSet = function (path, params) {
+      var width = params ? params['w'] : undefined;
+      var height = params ? params['h'] : undefined;
+      var aspectRatio = params ? params['ar'] : undefined;
+
+      // determines if an aspect ratio value is in the correct format 'w:h'
+      var isValidFormat = function() {
+        return /^\d+(\.\d+)?:\d+(\.\d+)?$/.test(aspectRatio);
+      }();
+
+      if (aspectRatio && !isValidFormat) {
+        throw new Error('The \'ar\' parameter key must follow the format w:h');
+      }
+
+      var fixedWidth = ((width && height) || (width && aspectRatio) || (height && aspectRatio)) ? true : false;
+
+      if (fixedWidth) {
+        return this._buildDPRSrcSet(path, params);
+      }
+      else {
+        return this._buildSrcSetPairs(path, params);
+      }
+    };
+
+    ImgixClient.prototype._buildSrcSetPairs = function(path, params) {
+      var srcset = '';
+      var targetWidths = function() {
```

Comment:
I think that `targetWidths` should definitely be pulled out into a separate file where the results are imported once into this main class. But with the way it's packaged together I'm not exactly sure what's the best way to do that.

Reply:
I agree, but I don't think it's worth messing around with the build system at the moment. I'd say for now just pull this out into a "constant" at the top of the file so it doesn't need to be recomputed every time this function is called.

```js
var TARGET_WIDTHS = (function() {
 // ...
})();

```